### PR TITLE
browser: Implement multi select user interface for object listings

### PR DESCRIPTION
**Pull request by rushenn**
## Description
Implement multi-select user interface for object listings. This is only the non-functional view part and need further work to make it functional. 

## Motivation and Context
https://github.com/minio/minio/issues/2854

## How Has This Been Tested?
Tested only the UI manually

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by rushenn** [comment]
Sample markup for action header:

action-header

```
<div className="list-actions">
      <span className="la-label">
           <i className="fa fa-check-circle"></i> 
           25 Objects selected
      </span>
      <span className="la-actions pull-right">
             <button>Download all as zip</button>
      </span>
</div>
```

**Comment by harshavardhana** [comment]
ping @rushenn @krishnasrinivas is this ready for review?

**Comment by rushenn** [comment]
@krishnasrinivas

**Comment by harshavardhana** [comment]
> @krishnasrinivas

@rushenn what else is pending here?

**Comment by rushenn** [comment]
@harshavardhana working on the functionality.

**Comment by krishnasrinivas** [comment]
checking

**Comment by krishnasrinivas** [comment]
@harshavardhana this has the html/less changes, now @rushenn is working on the JS/redux changes. We had a chat on how to go about the JS changes for zip/multi-delete issues.

File: .minio.sys/buckets/browser/policy.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"Version":"2012-10-17","Statement":[{"Action":["s3:GetBucketLocation"],"Effect":"Allow","Principal":{"AWS":["*"]},"Resource":["arn:aws:s3:::browser"],"Sid":""},{"Action":["s3:ListBucket"],"Condition":{"StringEquals":{"s3:prefix":["/docs","cmd","fefe fef "]}},"Effect":"Allow","Principal":{"AWS":["*"]},"Resource":["arn:aws:s3:::browser"],"Sid":""},{"Action":["s3:GetObject"],"Effect":"Allow","Principal":{"AWS":["*"]},"Resource":["arn:aws:s3:::browser//docs*","arn:aws:s3:::browser/cmd*","arn:aws:s3:::browser/fefe fef *"],"Sid":""}]}
```
Comment (harshavardhana) [inline .minio.sys/buckets/browser/policy.json]: You shouldn't add these files into your commit.  Also make sure to run minio on a separate directory.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by harshavardhana** [comment]
Tests are failing 

```
time="2017-02-20T22:25:32Z" level=error msg="Unable to parse JWT token string" cause="token contains an invalid number of segments" source="[jwt.go:113:isAuthTokenValid()]" 
--- FAIL: TestWebHandlerDownloadZip (0.26s)
	web-handlers_test.go:863: Expected to receive authentication error
```

File: browser/app/js/actions.js [hunk 2]
```diff
@@ -402,6 +404,24 @@ export const setLoginError = () => {
   }
 }
 
+export const downloadAllasZip = (url, req, xhr) => {
+  return (dispatch) => {
+    xhr.open('PUT', url, true)
```
Comment (harshavardhana) [inline browser/app/js/actions.js]: Why would this be a PUT request, it should be "GET"

File: cmd/web-router.go [hunk 3]
```diff
@@ -84,7 +84,7 @@ func registerWebRouter(mux *router.Router) error {
 	webBrowserRouter.Methods("POST").Path("/webrpc").Handler(webRPC)
 	webBrowserRouter.Methods("PUT").Path("/upload/{bucket}/{object:.+}").HandlerFunc(web.Upload)
 	webBrowserRouter.Methods("GET").Path("/download/{bucket}/{object:.+}").Queries("token", "{token:.*}").HandlerFunc(web.Download)
-	webBrowserRouter.Methods("GET").Path("/zip").Queries("token", "{token:.*}").HandlerFunc(web.DownloadZip)
```
Comment (harshavardhana) [inline cmd/web-router.go:87]: We should keep it as "GET" not PUT.  Get returns the reader not PUT.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: browser/app/js/actions.js (hunk 2 above)
Comment (rushenn) [reply browser/app/js/actions.js]: We couldn't send content using GET. @krishnasrinivas can give a better explanation on this

Comment (harshavardhana) [reply browser/app/js/actions.js]: in that case we should use POST instead of PUT,  PUT is not expected to do what is being done here.

Comment (harshavardhana) [reply browser/app/js/actions.js]: Also update the test case that is failing along with it.

**Comment by krishnasrinivas** [comment]
@harshavardhana after changing from PUT to POST it does not work for rushan (MacOS chrome) but works for me (linux chrome)
can you check if it works for you?

**Comment by krishnasrinivas** [comment]
> can you check if it works for you?

OK, it works for him too

**Comment by harshavardhana** [comment]
Odd that windows tests are failing, rebuilding them again.

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
There is some problem in the PR which is causing some problems on windows, not sure what would trigger this only change is POST handler.

**Comment by krishnasrinivas** [comment]
```
... obtained int = 411
... expected int = 200
```

411 is content-length required, weird
is triggering the test again a good idea? just in case it was some transient error.

**Comment by harshavardhana** [comment]
> is triggering the test again a good idea? just in case it was some transient error.

Already did it twice. I think we can pull this locally and see what is going on. Must be handlers are tripping over each other after adding POST>

**Comment by krishnasrinivas** [comment]
> Already did it twice. I think we can pull this locally and see what is going on. Must be handlers are tripping over each other after adding POST>
>  Show all reviewers

let me check on a windows machine

**Comment by krishnasrinivas** [comment]
@rushenn can you rebase and push it again? the windows appveyor issue is fixed in master.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**